snowpark-checkpoints-hypothesis/src/snowflake/hypothesis_snowpark/strategies_utils.py

```diff
@@ -118,19 +127,34 @@ def apply_custom_null_values(
 
     for column, target_null_proportion in null_proportions.items():
         if column not in df_copy.columns:
+            LOGGER.warning(
+                "Column '%s' not found in DataFrame during null value correction. Skipping it.",
+                column,
+            )
             continue
 
         required_nulls = round(total_rows * target_null_proportion)
         current_nulls = df_copy[column].isnull().sum()
 
         if current_nulls < required_nulls:
+            LOGGER.debug(
+                "Column '%s' has %d null values but %d are required.",
```

Comment:
Please replace the %d with %s
```suggestion
                "Column '%s' has %s null values but %s are required.",
```

Reply:
Fixed in [c89f432](https://github.com/snowflakedb/snowpark-checkpoints/pull/120/commits/c89f43245f4eac95152c600b0e8f2043d21262ed)